geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -470,10 +470,28 @@ public static int javaImplementationOfAnsiCMemCmp(byte[] array1, byte[] array2)
     // with same name...
   }
 
-  static class OrderedSetEntry implements Comparable<OrderedSetEntry> {
-    private final byte[] member;
-    private final byte[] scoreBytes;
-    private final Double score;
+  abstract static class AbstractOrderedSetEntry implements Comparable<AbstractOrderedSetEntry> {
```

Comment:
I like this. One small suggestion would be to pull `compareTo` into this class but declare a separate `abstract` method that is the only abstraction difference in subclasses:
```
    protected abstract int internalCompare(byte[] b1, byte[] b2);

    @Override
    public int compareTo(AbstractOrderedSetEntry o) {
      int comparison = score.compareTo(o.score);
      if (comparison == 0) {
        // Scores equal, try lexical ordering
        comparison = internalCompare(member, o.member);
      }
      return comparison;
    }
  }
```

Reply:
Ah, good call, that's neater.